docroot/modules/humsci/hs_config_readonly/src/EventSubscriber/ConfigReadOnlyEventSubscriber.php

```diff
@@ -0,0 +1,222 @@
+      $entity = $form_object->getEntity();
+      $mark_form_read_only = $entity instanceof ConfigEntityInterface;
+    }
+
+    // Don't block particular patterns.
+    if ($mark_form_read_only && $form_object instanceof EntityFormInterface) {
+      $entity = $form_object->getEntity();
+      $name = $entity->getConfigDependencyName();
+
+      // Block config from a module.
+      $mark_form_read_only = $this->configIsEditable([$name]);
+
+      // Unless its defined as a whiteist pattern.
+      if ($this->matchesWhitelistPattern($name)) {
+        $mark_form_read_only = FALSE;
+      }
+    }
+
+    // Config forms.
+    if ($mark_form_read_only && $form_object instanceof ConfigFormBase) {
+      $editable_config = $this->getEditableConfigNames($form_object);
+      $mark_form_read_only = $this->configIsEditable($editable_config);
+    }
+
+    if ($mark_form_read_only) {
+      $event->markFormReadOnly();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events = [];
+    $events[ReadOnlyFormEvent::NAME][] = ['onFormAlter', 200];
+    return $events;
+  }
+
+  /**
+   * Get the editable configuration names.
+   *
+   * @param ConfigFormBase $form
+   *   The configuration form.
+   *
+   * @return array
+   *   An array of configuration object names that are editable if called in
+   *   conjunction with the trait's config() method.
+   *
+   * @throws \ReflectionException
+   */
+  protected function getEditableConfigNames(ConfigFormBase $form) {
+    // Use reflection to work around getEditableConfigNames() as protected.
+    // @todo Review in 9.x for API change.
+    // @see https://www.drupal.org/node/2095289
+    $reflection = new \ReflectionMethod(get_class($form), 'getEditableConfigNames');
+    $reflection->setAccessible(TRUE);
+    return $reflection->invoke($form);
+  }
+
+  /**
+   * Check if the given configuration name can be edited.
+   *
+   * @param array $config
+   *   Array of config names.
+   *
+   * @return bool
+   *   If it can be edited.
+   */
+  protected function configIsEditable(array $config) {
+    if ($config == array_filter($config, [$this, 'isEditableConfig'])) {
+      return FALSE;
+    }
+    return TRUE;
+  }
+
+  /**
+   * @param $name
+   *
+   * @return bool
+   */
+  protected function isEditableConfig($name) {
+    foreach ($this->getLockedConfigs() as $config) {
+      if ($config == $name) {
+        return FALSE;
+      }
+    }
+    return TRUE;
+  }
+
+  /**
+   * Get all configs provided by modules.
+   *
+   * @return array
+   *   Config names.
+   */
+  protected function getLockedConfigs() {
+    $module_configs = [];
+    foreach ($this->moduleHandler->getModuleList() as $name => $extension) {
+      if ($extension->getType() == 'profile' || in_array($name, $this->excludedModules)) {
+        continue;
+      }
+
+      $install_list = $this->listProvidedItems($name);
+      $optional_list = $this->listProvidedItems($name, TRUE);
```

Comment:
🤔 Are you sure you want to lock the optional ones as well?

Reply:
I figured i'd treat them the same as install ones. since the optional ones could consist of fields that rely on other content types. like a entity reference field.

Reply:
Would it be possible that you could get in to a scenario that would prevent the user from configuring a new module that was turned on through the UI because it's optional config is now available?

Reply:
well right now modules can only be turned on via drush. and yes it would prevent that. maybe we limit the modules/profiles that are "blacklisted"?

Reply:
Yeah, I think this will massage itself out over time. Keep this scenario in mind when you get a user that is blocked from new config screens. I am sure we would want to define a scenario or have a list of items to whitelist.